### PR TITLE
Fix undefined class name in Table component

### PR DESCRIPTION
**Pull request by meckila-navapbc**
Connected to #16005

### Description
Account for case where className is undefined in Table component. Currently returns undefined

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] Jest tests pass
- [ ] PropTypes are added

### CODE REVIEW

File: client/app/components/Table.jsx [hunk 1]
```diff
@@ -289,7 +325,7 @@ Table.propTypes = {
     PropTypes.func]).isRequired,
   rowObjects: PropTypes.arrayOf(PropTypes.object).isRequired,
   rowClassNames: PropTypes.func,
-  keyGetter: PropTypes.func,
```
Comment (meckila-navapbc) [inline client/app/components/Table.jsx:292]: This wasn't a prop

File: client/app/queue/QueueTable.jsx [hunk 2]
```diff
@@ -626,7 +626,7 @@ export default class QueueTable extends React.PureComponent {
         <HeaderRow
           columns={columns}
           rowObjects={rowObjects}
-          headerClassName={headerClassName}
+          headerClassName={headerClassName ?? ''}
```
Comment (jcq) [inline client/app/queue/QueueTable.jsx:629]: This is fine, but I think the implementation of `HeaderRow` means that we would actually also be OK w/ `undefined` here; unlike how `className` was being handled in `Table`, this isn't dropping the expression into a string literal.

File: client/app/components/Table.jsx [hunk 3]
```diff
@@ -252,7 +252,7 @@ export default class Table extends React.PureComponent {
 
     return <table
       id={id}
-      className={`usa-table-borderless ${this.props.className}`}
+      className={`usa-table-borderless ${this.props.className ?? ''}`}
```
Comment (jcq) [inline client/app/components/Table.jsx:255]: 🎉

File: client/app/components/Table.jsx [hunk 4]
```diff
@@ -281,6 +281,42 @@ export default class Table extends React.PureComponent {
   }
 }
 
+HeaderRow.propTypes = {
```
Comment (jcq) [inline client/app/components/Table.jsx:284]: hooray for new proptypes!

File: client/app/components/Table.jsx [hunk 5]
```diff
@@ -289,7 +325,7 @@ Table.propTypes = {
     PropTypes.func]).isRequired,
   rowObjects: PropTypes.arrayOf(PropTypes.object).isRequired,
   rowClassNames: PropTypes.func,
-  keyGetter: PropTypes.func,
+  getKeyForRow: PropTypes.func,
```
Comment (jcq) [inline client/app/components/Table.jsx:328]: double hooray for the _correct_ proptypes. 😆

**Review by jcq — APPROVED** [review]
👍 

Thanks for adding in the missing proptypes for the subcomponents too

File: client/app/queue/QueueTable.jsx (hunk 2 above)
Comment (meckila-navapbc) [reply client/app/queue/QueueTable.jsx:629]: I can change this back! I thought that is was being applied the same way as `Table`

**Pull request merged**